core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -175,6 +175,13 @@ class Partition(val topic: String,
       // add replicas that are new
       allReplicas.foreach(replica => getOrCreateReplica(replica))
       val newInSyncReplicas = partitionStateInfo.isr.asScala.map(r => getOrCreateReplica(r)).toSet
+
+      // If the ISR size received from a LeaderAndIsrRequest is smaller than the existing ISR size,
+      // mark the isrShrinkRate meter. For example: This can happen if one of the followers is shutdown.
+      // Note that this metric is not triggered on follower -> leader transition
+      // because the inSyncReplicas.size is always zero on a follower
+      if (newInSyncReplicas.size < inSyncReplicas.size) replicaManager.isrShrinkRate.mark()
```

Comment:
That is correct - I don't think we can fix this easily and as I mentioned above I think it would be better to just doc the shrink rate meter on what shrinks it can and cannot record. Alternately we could record a shrink in the leader to follower transition (since the controller would send the new shrunk ISR) but I don't feel that is worth it.